Firestore/core/src/firebase/firestore/remote/serializer.h

```diff
@@ -226,6 +226,29 @@ class Serializer {
 
   std::string EncodeQueryPath(const model::ResourcePath& path) const;
 
+  void ValidateDocumentKeyPath(nanopb::Reader* reader,
```

Comment:
This is my attempt to refactor the common part from `DecodeKey` and the newly-added `DecodeDatabaseId`.